modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -2240,6 +2244,73 @@ public void sendDeferredAckResponse(UUID nodeId, GridCacheVersion ver) {
     }
 
     /**
+     * Detaches current thread from the transaction.
+     *
+     * @param tx Transaction to be detached.
+     * @return {@code False} if already detached.
+     */
+    public void suspendTx(final GridNearTxLocal tx) throws IgniteCheckedException {
+        assert tx != null;
+        assert !tx.system() && !tx.pessimistic();
+
+        if (!tx.suspendInProgress())
+            throw new IgniteCheckedException("Use suspendTx prohibited. Use tx.suspend() instead.");
+
+        clearThreadMap(tx);
```

Comment:
As far as I can see other methods use clearThreadMap.
Moreover common method for clearing allows some additional checks and actions inside it.